RInAppMessaging/Classes/Services/ConfigurationService.swift

```diff
@@ -35,6 +36,8 @@ internal struct ConfigurationService: ConfigurationServiceType, HttpRequestable
             switch requestError {
             case .httpError(let statusCode, _, _) where statusCode == 429:
                 return .failure(.tooManyRequestsError)
+            case .httpError(let statusCode, _, _) where [404, 400].contains(statusCode):
+                return .failure(.missingOrInvalidSubscriptionId)
```

Comment:
Is missing/invalid subId the only scenario where we could get 404, 400 status codes?

Reply:
Actually I think I'll separate them. because 'missing'(400) doesn't mean that subscription id 'doesn't exist' (404)